packages/react-charting/src/components/SankeyChart/SankeyChart.base.tsx

```diff
@@ -1,26 +1,87 @@
+
 const getClassNames = classNamesFunction<ISankeyChartStyleProps, ISankeyChartStyles>();
+const PADDING_PERCENTAGE = 0.3;
 
-export class SankeyChartBase extends React.Component<
-  ISankeyChartProps,
-  {
-    containerWidth: number;
-    containerHeight: number;
-  }
-> {
+type NodesInColumns = { [key: number]: SNode[] };
+type NodeColors = { fillColor: string; borderColor: string };
+
+export interface ISankeyChartState extends IBasestate {
+  containerWidth: number;
+  containerHeight: number;
+  selectedState: boolean;
+  selectedLinks: Set<Number>;
+  selectedNodes: Set<Number>;
+  selectedNode?: SNode;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  nodeElement?: any;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  linkElement?: any;
+  selectedLink?: SLink;
+  shouldOverflow: boolean;
+}
+
+const NON_SELECTED_NODE_AND_STREAM_COLOR: string = '#757575';
+const DEFAULT_NODE_COLORS: NodeColors[] = [
+  { fillColor: '#00758F', borderColor: '#002E39' },
+  { fillColor: '#77004D', borderColor: '#43002C' },
+  { fillColor: '#4F6BED', borderColor: '#3B52B4' },
+  { fillColor: '#937600', borderColor: '#6D5700' },
+  { fillColor: '#286EA8', borderColor: '#00457E' },
+  { fillColor: '#A43FB1', borderColor: '#7C158A' },
+  { fillColor: '#CC3595', borderColor: '#7F215D' },
+  { fillColor: '#0E7878', borderColor: '#004E4E' },
+  { fillColor: '#8764B8', borderColor: '#4B3867' },
+  { fillColor: '#9C663F', borderColor: '#6D4123' },
+];
+
+const MIN_HEIGHT_FOR_DOUBLINE_TYPE = 36;
+const MIN_HEIGHT_FOR_TYPE = 24;
+const REST_STREAM_OPACITY: number = 1;
+const NON_SELECTED_OPACITY: number = 1;
+const SELECTED_STREAM_OPACITY: number = 0.3;
+const NON_SELECTED_STREAM_BORDER_OPACITY: number = 0.5;
+const DEFAULT_TEXT_COLOR: string = '#323130';
+const NON_SELECTED_TEXT_COLOR: string = '#FFFFFF';
+
+export class SankeyChartBase extends React.Component<ISankeyChartProps, ISankeyChartState> {
   private _classNames: IProcessedStyleSet<ISankeyChartStyles>;
   private chartContainer: HTMLDivElement;
   private _reqID: number;
+  private _calloutId: string;
+  private _linkId: string;
+  private _nodesInColumn: NodesInColumns;
+  private _sankey: d3Sankey.SankeyLayout<d3Sankey.SankeyGraph<{}, {}>, {}, {}>;
+  private _margins: IMargins;
+
   constructor(props: ISankeyChartProps) {
     super(props);
     this.state = {
-      containerHeight: 0,
-      containerWidth: 0,
+      containerHeight: 468,
+      containerWidth: 912,
+      selectedState: false,
+      selectedLinks: new Set<number>(),
+      selectedNodes: new Set<number>(),
+      shouldOverflow: false,
+      isCalloutVisible: false,
     };
+    this._calloutId = getId('callout');
+    this._linkId = getId('link');
+    this._margins = { top: 36, right: 48, bottom: 32, left: 48 };
+    this._preRenderLayout();
+    this._nodesInColumn = this._populateNodeInColumns(this._sankey);
+    this._adjustOnePercentHeightNodes(this._nodesInColumn);
```

Comment:
same for these 2 steps. Should they be part of a different UI lifecycle call

Reply:
This is also fixed .